detox/src/devices/runtime/drivers/ios/XCUITestUtils.js

```diff
@@ -0,0 +1,223 @@
+    TEST_RUNNER_DETOX_DISABLE_VIEW_HIERARCHY_DUMP: disableDumpViewHierarchy
+  };
+
+  const options = {
+    maxBuffer: 1024 * 1024 * 10, // 10MB
+  };
+
+  const command = `${env.TEST_RUNNER_IS_DETOX_ACTIVE ? Object.keys(env).map(key => `${key}=${env[key]}`).join(' ') : ''} ${xcodebuildBinary} ${flags.map(flag => flag.includes(' ') ? `"${flag}"` : flag).join(' ')}`;
+  log.debug(`[XCUITest] Running command: ${command}`);
+
+  const isRunningOnTerminal = process.stdout.isTTY;
+  if (isRunningOnTerminal !== true) {
+    log.debug(`[XCUITest] Currently not running through the Terminal, trying to execute the command from the Terminal`);
+
+    try {
+      return _runCommandInTerminal(command, options);
+    } catch (e) {
+      log.warn(`[XCUITest] Failed to execute the command from the Terminal, falling back to the default execution`);
+    }
+  }
+
+  const result = exec(command, options);
+
+  result.childProcess.stderr.on('data', (data) => {
+    if (data.includes('Testing started completed')) {
+      result.childProcess.kill('SIGTERM');
+    }
+  });
+
+  return result;
+}
+
+function _runCommandInTerminal(command, options) {
+  const escapedCommand = command.replace(/"/g, '\\"');
+
+  // Opens the Terminal app and runs the command in a new window, then closes the window when the command is done running.
+  const appleScript = `
+        tell application "Terminal"
+            if not running then
+                open
+            end if
+
+            activate
+            do script "${escapedCommand}; exit"
+
+            repeat
+                delay 1
+                if not busy of window 1 then
+                    close window 1
+                    exit repeat
+                end if
+            end repeat
+        end tell
+    `;
+
+  return exec(`osascript -e '${appleScript}'`, options);
+}
+
+function _allowNetworkPermissionsXCUITest() {
+  log.debug(`[XCUITest] Allowing network permissions`);
+  let didCallback = false;
+
+  const childProcess = osascript.executeFile(
+    `${__dirname}/allowNetworkPermissionsXCUITest.scpt`,
+    function(err, _, __) {
+      if (err) {
+        log.error(`[XCUITest] Failed to approve network permissions for XCUITest target:\n\t${err}`);
+      } else {
+        log.debug(`[XCUITest] Network permissions are allowed`);
+      }
+
+      didCallback = true;
+    });
+
+  setTimeout(() => {
+    if (didCallback || !childProcess) {
+      return;
+    }
+
+    log.debug(`[XCUITest] Killing the process that allows network permissions (timed-out)`);
+    childProcess.stdin.pause();
+    childProcess.kill('SIGTERM');
+
+  }, 30000).unref();
+}
+
+async function _waitForTestTargetServerToStart(testTargetServerPort, cpPromise) {
+  log.debug(`[XCUITest] Waiting for test target server to start on port ${testTargetServerPort}...`);
+  let isServerUp = false;
+  let retries = 0;
+
+  while (!isServerUp && retries++ < 90) {
+    try {
+      await exec(`nc -z localhost ${testTargetServerPort}`);
```

Comment:
This looks too heavy. You could probe a port using `net`.Socket package, I guess.